src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java

```diff
@@ -61,7 +61,7 @@ private PermanentStorage createStorageInstance(String type) throws SapsException
         if (lType.equals("swift"))
             return new SwiftPermanentStorage(properties);
         if (lType.equals("nfs"))
-            return new NfsPermanentStorage();
+            return new NfsPermanentStorage(properties);
```

Comment:
maybe, the Archiver could receive a reference to the `PermanentStorage` in the interface

Reply:
... instead of parsing the config file and creating it.